### DISCUSSION TITLE
How to use node plugin version of nodejs and npm?

### DISCUSSION THREAD

**Discussion by FSeidinger-XI** [node-gradle/gradle-node-plugin#327] [Q&A]
Let's say we take the following assumptions:

1. I have a system wide `node` and `npm` version installed with some older version.
2. I have setup the `node` plugin to install a newer version of `node` and `npm`.
3. After installation the versions are found in `.gradle/nodejs/nodejs-some_version` and `.gradle/npm/npm-some_version`.

Is there a trick or recipe to use this versions if running `node` or `npm` on the command line?

Background is that when running, testing or building on the command line I want to be sure to use the same version that gradle sees.

I could alter my `PATH` variable but then it includes the version and will not change, if I upgrade the versions in gradle.

**Comment by deepy** [comment]
There's a script in one of the issues for doing this, and there's an issue that's requesting support for NVM
But there's nothing preventing the opposite direction :-)

You could have a task that writes a [.nvmrc](https://github.com/nvm-sh/nvm?tab=readme-ov-file#nvmrc) with the version specified in Gradle and use nvm's shell integration for that
